src/components/PeopleTable.tsx

```diff
@@ -1,682 +1,141 @@
-export const PeopleTable = () => {
+import cn from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[],
+};
+
+const columns = [
+  { Name: 'name' },
+  { Sex: 'sex' },
+  { Born: 'born' },
+  { Died: 'died' },
+];
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const [searchParams] = useSearchParams();
+  const sortValue = searchParams.get('sort');
+  const sortOrder = searchParams.get('order');
+
+  const filterPeople = (p: Person[]) => {
+    let peopleCopy = [...p];
+    const filterBySex = searchParams.get('sex');
+    const filterByQuery = searchParams.get('query');
+    const filterByCentury = searchParams.getAll('centuries');
+
+    if (filterBySex) {
+      peopleCopy = peopleCopy.filter((person => (person.sex === filterBySex)));
+    }
+
+    if (filterByQuery) {
+      peopleCopy = peopleCopy.filter(
+        input => input.name.toLowerCase().includes(filterByQuery)
+        || input.motherName?.toLowerCase().includes(filterByQuery)
+        || input.fatherName?.toLowerCase().includes(filterByQuery),
+      );
+    }
+
+    if (filterByCentury && filterByCentury.length > 0) {
+      peopleCopy = peopleCopy.filter(person => {
+        return (
+          filterByCentury?.includes(Math.ceil(person.born / 100).toString())
+        );
+      });
+    }
+
+    if (sortValue) {
+      peopleCopy.sort((a, b) => {
+        switch (sortValue) {
+          case ('name'):
+            return a.name.localeCompare(b.name);
+          case ('sex'):
+            return a.sex.localeCompare(b.sex);
+          case ('born'):
+            return (a.born - b.born);
+          case ('died'):
+            return (a.died - b.died);
+          default: return 0;
+        }
+      });
+    }
```

Comment:
you can simplify this a little, for example, name and sex have the same logic for sort so you can write
`case ('name'):
case ('sex'):
 return a[sortValue].localeCompare(b[sortValue]);`